test/process_test.cc

```diff
@@ -26,11 +26,11 @@ class ProcessTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
       : loopback_address_(Envoy::Network::Test::getLoopbackAddressUrlString(GetParam())),
         options_(TestUtility::createOptionsImpl(
             fmt::format("foo --duration 1 -v error --rps 10 https://{}/", loopback_address_))){};
-  void runProcess() {
+  void runProcess(bool expect_success = true) {
```

Comment:
Consider doing enum class { EXPECT_SUCCESS, EXPECT_FAILURE} which makes the code easier to read without comments.